### PR TITLE
chore: add documentation for model-hub mmdetection [DET-5924]

### PR DESCRIPTION
**Pull request by liamcli**
## Description
Add documentation for model-hub mmdetection.  

Also fixed pytorch and torchvision versions that were out-of-date in circleci config so that they now match the versions for our default gpu image.  

Release note 2792 covers this PR as well.

## Test Plan
Made sure e2e tests passed in feature branch.  

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [x] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](../docs/release-notes/README.md) for details.
- [x] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: model_hub/model_hub/mmdetection/_trial.py [hunk 1]
```diff
@@ -81,7 +91,6 @@ def __init__(self, context: det_torch.PyTorchTrialContext) -> None:
             mmcv.runner.build_optimizer(self.model, self.cfg.optimizer)
         )
         self.model.zero_grad()
-        self.optimizer.zero_grad()
```
Comment (shiyuann) [inline model_hub/model_hub/mmdetection/_trial.py:84]: Is it ok to just remove this?

File: model_hub/examples/mmdetection/fasterrcnn.yaml [hunk 2]
```diff
@@ -15,35 +15,35 @@ data:
...
   max_length:
-    batches: 87850
+    batches: 87960
   smaller_is_better: false
-max_restarts: 0
+max_restarts: 5
 environment:
   image:
-    gpu: determinedai/model-hub-mmdetection
+    gpu: determinedai/model-hub-mmdetection:0.17.0.dev0
```
Comment (shiyuann) [inline model_hub/examples/mmdetection/fasterrcnn.yaml:44]: Should use commit id.

**Review by shiyuann — COMMENTED** [review]
I don't have too much context on what needs to be supported in the model hub. The change in the documentation and CI looks good.

File: model_hub/examples/mmdetection/fasterrcnn.yaml (hunk 2 above)
Comment (liamcli) [reply model_hub/examples/mmdetection/fasterrcnn.yaml:44]: we push tip of master to the dev tag every commit.  on release, the dev suffix is removed.  the assumption is that users will checkout a release tag or just download the example from our doc.

File: model_hub/model_hub/mmdetection/_trial.py (hunk 1 above)
Comment (liamcli) [reply model_hub/model_hub/mmdetection/_trial.py:84]: optimizer parameters are a subset of model parameters so model.zero_grad() will cover everything.

File: model_hub/examples/mmdetection/fasterrcnn.yaml (hunk 2 above)
Comment (shiyuann) [reply model_hub/examples/mmdetection/fasterrcnn.yaml:44]: > on release, the dev suffix is removed.

This won't be done automatically. Is the plan to remove the tag?

Comment (liamcli) [reply model_hub/examples/mmdetection/fasterrcnn.yaml:44]: This is done automatically as part of bumpenvs run during release process. It's one reason we need the dev version here.

**Review by shiyuann — APPROVED** [review]
